apps/astarte_housekeeping/lib/astarte_housekeeping/engine.ex

```diff
@@ -192,6 +223,22 @@ defmodule Astarte.Housekeeping.Engine do
     end
   end
 
+  defp delete_datastream_maximum_storage_retention(realm_name) do
+    case Queries.delete_datastream_maximum_storage_retention(realm_name) do
+      {:ok, %Xandra.Void{}} ->
+        :ok
+
+      {:error, reason} ->
+        _ =
+          Logger.warning(
+            "Cannot delete datastream maximum storage retention, error #{inspect(reason)}",
+            tag: "delete_datastream_maximum_storage_retention"
+          )
+
+        {:error, :delete_datastream_maximum_storage_retention_fail}
```

Comment:
These are new errors so you have to handle them in Housekeeping APC in `extract_reply` in the `RPC` module and in `FallbackController`, otherwise an Internal Server Error is returned instead of a more informative error